provider/datagen/src/transform/cldr/plurals/mod.rs

```diff
@@ -3,87 +3,48 @@
 }
 
 impl From<&SourceData> for PluralsProvider {
     fn from(source: &SourceData) -> Self {
         PluralsProvider {
             source: source.clone(),
-            cardinal_rules: RwLock::new(None),
-            ordinal_rules: RwLock::new(None),
         }
     }
 }
 
 impl PluralsProvider {
-    fn get_rules_for(
-        &self,
-        key: ResourceKey,
-    ) -> Result<std::sync::RwLockReadGuard<Option<Option<cldr_serde::plurals::Rules>>>, DataError>
-    {
-        Ok(match key {
-            CardinalV1Marker::KEY => {
-                #[allow(clippy::unwrap_used)]
-                // TODO(#1668) Clippy exceptions need docs or fixing.
-                if self.cardinal_rules.read().expect("poison").is_none() {
-                    let path = self
-                        .source
-                        .get_cldr_paths()?
-                        .cldr_core()
-                        .join("supplemental")
-                        .join("plurals.json");
-                    let data: cldr_serde::plurals::Resource =
-                        serde_json::from_reader(open_reader(&path)?)
-                            .map_err(|e| DataError::from(e).with_path_context(&path))?;
-                    let _ = self
-                        .cardinal_rules
-                        .write()
-                        .expect("poison")
-                        .get_or_insert(data.supplemental.plurals_type_cardinal);
-                }
-                #[allow(clippy::unwrap_used)]
-                // TODO(#1668) Clippy exceptions need docs or fixing.
-                self.cardinal_rules.read().expect("poison")
-            }
-            OrdinalV1Marker::KEY => {
-                #[allow(clippy::unwrap_used)]
-                // TODO(#1668) Clippy exceptions need docs or fixing.
-                if self.ordinal_rules.read().expect("poison").is_none() {
-                    let path = self
-                        .source
-                        .get_cldr_paths()?
-                        .cldr_core()
-                        .join("supplemental")
-                        .join("ordinals.json");
-                    let data: cldr_serde::plurals::Resource =
-                        serde_json::from_reader(open_reader(&path)?)
-                            .map_err(|e| DataError::from(e).with_path_context(&path))?;
-                    let _ = self
-                        .ordinal_rules
-                        .write()
-                        .expect("poison")
-                        .get_or_insert(data.supplemental.plurals_type_ordinal);
-                }
-                #[allow(clippy::unwrap_used)]
-                // TODO(#1668) Clippy exceptions need docs or fixing.
-                self.ordinal_rules.read().expect("poison")
-            }
-            _ => return Err(DataError::custom("Unknown key for PluralRulesV1")),
-        })
+    fn get_rules_for(&self, key: ResourceKey) -> Result<&cldr_serde::plurals::Rules, DataError> {
```

Comment:
Praise: This function is much nicer. We can do this now with the centralized cache.